File: app/src/main/java/br/com/jadlog/signature/FinallyActivity.java
package br.com.jadlog.signature;

import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.view.KeyEvent;
import android.widget.ImageView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.widget.Toolbar;

public class FinallyActivity extends CordovaActivity {
    private Toolbar toolbar;
    private ActionBar actionBar;
    private ImageView imageView;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.finally_activity);

        final byte[] hash   = getIntent().getByteArrayExtra("HASH");
        final Bitmap bitmap = BitmapFactory.decodeByteArray(hash, 0, hash.length);

        toolbar   = findViewById(R.id.toolbar);
        imageView = findViewById(R.id.imageView);
        imageView.setImageBitmap(bitmap);

        actionBar();
    }

    private void actionBar() {
        // toolbar
        setSupportActionBar(toolbar);
        actionBar = getSupportActionBar();

        if (actionBar != null) {
            actionBar.setTitle("");
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setHomeButtonEnabled(true);
        }
    }

    private void back() {
        startActivity(new Intent(this, SignatureActivity.class));
        finish();
        overridePendingTransition( R.anim.lefttoright, R.anim.stable );
    }

    @Override
    public boolean onSupportNavigateUp() {
        back();
        return true;
    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {
        if (keyCode == KeyEvent.KEYCODE_BACK) { back(); return false; }
        else                                  { return super.onKeyDown(keyCode, event); }
    }
}

File: app/src/main/java/br/com/jadlog/signature/view/FinallyActivity.kt
package br.com.jadlog.signature.view

import android.content.Intent
import android.graphics.BitmapFactory
import android.view.KeyEvent
import androidx.appcompat.app.ActionBar
import kotlinx.android.synthetic.main.finally_activity.*

import br.com.jadlog.signature.R

class FinallyActivity : BaseActivity(R.layout.finally_activity) {
    override fun initViews() {
        val hash = intent.getByteArrayExtra("HASH")
        val bitmap = BitmapFactory.decodeByteArray(hash, 0, hash.size)

        imageView.setImageBitmap(bitmap)

        actionBar()
    }

    override fun initViewModel() {
    }

    private fun actionBar() {
        setSupportActionBar(findViewById(R.id.toolbar))

        val actionBar: ActionBar = supportActionBar!!
            actionBar.title = ""
            actionBar.setDisplayHomeAsUpEnabled(true)
            actionBar.setHomeButtonEnabled(true)
    }

    private fun back() {
        startActivity(Intent(this, MainActivity::class.java))
        finish()
        overridePendingTransition(R.anim.lefttoright, R.anim.stable)
    }

    override fun onSupportNavigateUp(): Boolean {
        back()
        return true
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {
        return if (keyCode == KeyEvent.KEYCODE_BACK) {
            back()
            false
        }
        else {
            super.onKeyDown(keyCode, event)
        }
    }
}